### DISCUSSION TITLE
Mechanic and hull  repairs improvement

### DISCUSSION THREAD

**Discussion by Swato85** [FakeFishGames/Barotrauma#7403] [Ideas and suggestions]
A way to make hull repairs and  mechanic a more important things.
1. Make Hull repairs (maybe depending on the hole size)  require materials for repair, not only welding tool.
- make hole repairs more immersive by making the torn out pieces require actual material to be repaired.
- any of the metals to be  used when repairing. Whoever is doing the repairs should have it in inventory and it will get consumed depending on the  repairs.
- if no materials are available, the hull  may be  repairable, but quite slow.
- you could also deconstruct equipment on sub in order to get materials for hull repairs. the scenario would quite fit the setting.
Mechanic role will become more important for its effective use of materials.
- new  talents may involve minimal or no material use for repairs.

another alternative is for the mechanic to be able to produce somekind of patching plates which  would be used instead of direct materials mechanic. Being more effective and easier to apply.

It would be more immersive if the crew was welding actual material to seal  the  bigger holes as it looks more logical than to recreate hull by a welding tool. The  nanobot  welding tool fits less into the world than sailors welding any materials they find to  patch missing hull pieces.

**Comment by Quartofel** [comment]
Consuming materials would be nice, especially that there are mats have "percentage". Catastrophic hull breaches don't feel too intimidating after you realise that you only need to swipe welder twice in the general direction of the breach to fix the problem.

**Comment by MaaiG** [comment]
Any event can be turned into a problem solved through a complex quest. But then the probability of the event itself must be minimized. Now, monsters eating through shell like a cardboard box! Then you need to amplify the shell or parts of it? Otherwise, all you’re going to be looking for is material to make a material for fixing material. And so in every specialty...

**Reply by Swato85** [reply]
> Any event can be turned into a problem solved through a complex quest. But then the probability of the event itself must be minimized. Now, monsters eating through shell like a cardboard box! Then you need to amplify the shell or parts of it? Otherwise, all you’re going to be looking for is material to make a material for fixing material. And so in every specialty...

I think  that player should not be able to simply shrug off beasts tearing the sub. There are gunners, and captain. Either they should be shot or maneuvering should be at play,  not tanking everything because repairs  are so easy currently.

**Comment by mLuby** [comment]
There's a mod called [Repair++](https://steamcommunity.com/sharedfiles/filedetails/?id=2236039375) that does this. I like the concept, though some people said it was a pain to play with this "need parts for repairs" mechanic.

The trouble is that a hull breach or electrical failure when you have no welding fuel or fuses left is a **soft game over** in a very slow, frustrating way. I'm usually not a fan of that: it'd steepen the difficulty and learning curves for an already complex and under-explained game. But maybe there's a solution, like very slow-to-make "improvised" repair parts that don't let the machine operate at its full capacity till you get proper parts.

**Reply by Swato85** [reply]
Well that also part of the point, the marines should be aware of possible dangers, not just speed into enemy because the crysis  management is simply waving around with a welder, it simply takes away the importance of teh  sub shell and threat assesment.

**Reply by MerelyMezz** [reply]
Repair++ Author here. The mod was made to add an element of resource management to the reparing duty, something that I feel adds more interesting complexity to how repairs are done, without resorting to shallow minigames (which I've always thought were an awful way of "improving" repairs).

In regards to hulls, we already have a resource that is being drained when they are repaired, which is welding fuel. You could maybe make the argument that it's maybe too plentiful and should be harder to aquire or faster to drain, or that welding as a whole should be much slower, but I don't believe that adding a secondary resource ontop of that is a good idea. You would essentially create a bottleneck situation, where you need BOTH a supply of welding fuel AND whatever hull materials, which can lead much more easily to what mLuby refers to as a softlock.

IMO if you wanted to improve the importance of hull repairs, the way to do it is by making changes to how quickly hull can be damaged or repaired entirely. The idea should be to encourage welding small parts of damage early on and as soon as possible, to prevent larger breaches. A huge hole should take quite a while to fix and present a serious threat to the submarine as a whole, requiring flood containment measures and alot of attention. This would also more effectively "fix" immersion issues, because if welding takes a much slower and "realistic" time to do, you probably don't really mind that you are still essentially fixing the wall out of thin air, as long as it *looks* plausible.

**Reply by Swato85** [reply]
Exactly, the importance of tending to smaller issues should be taken into account and biggger breaches should be feared. Secondary resource management is not  an issue. There is  a lot of everything everywhere, either through mining or buying it, the correct issue with missing materials is as simple as it sounds - unattended logistics and resourcing. On top of that, I also suggested that  in case  of missing materials the repairs could simply take longer for example, or even better maybe to be limited to how much hull strength is regained. It would give a proper rugged look to this space when you look at  subs and see shaded or otherwise discolored areas, marks  of old battles not yet completely fixed. This  parts are not of full strength  and can be penetrated/broken more easily. Get some materials together and reinforce it and again a fine area for a mechanic to  show his tricks. Minigames, although I would prefer classtasks or sub-tasks as a name,  add pressure and brains to things, they can be made interesting andbring value to a lot of things. The situations like mechanic telling what materials he needs to create some broken thingy in order to repair  some device to either  give it permanent boost or repair its current debuff. Simply living  the  world as fits the setting. people cry about having nothing to do, but when you  suggest something they immediately react that  its too much  work, lol simply  like most lazy workers.

**Reply by Swato85** [reply]
just  camed to mind, imagine the game Among them without minitasks, just  with timers or  some left-right simpleton strikes, It would definitely take away concentration/brains  and kill most of the feelings.
